src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -3033,13 +3051,17 @@ makeLimit (conn, _) (Limit ml mo) =
         v = maybe 0 fromIntegral
     in (TLB.fromText limitRaw, mempty)
 
-makeLocking :: LockingClause -> (TLB.Builder, [PersistValue])
-makeLocking = flip (,) [] . maybe mempty toTLB . Monoid.getLast
-  where
-    toTLB ForUpdate           = "\nFOR UPDATE"
-    toTLB ForUpdateSkipLocked = "\nFOR UPDATE SKIP LOCKED"
-    toTLB ForShare            = "\nFOR SHARE"
-    toTLB LockInShareMode     = "\nLOCK IN SHARE MODE"
+makeLocking :: IdentInfo -> LockingClause -> (TLB.Builder, [PersistValue])
+makeLocking info lockingClause =
+  case Monoid.getLast lockingClause of
```

Comment:
This monoid behavior is disappointing. It renders composability of locking clauses pointless.

Since we're doing a major bump for this change, we can change the `Monoid` instance to select the *most restrictive* of two options.

I'm not sure about the proper way to do that, so it's fine to punt that to a separate PR and require it for a 3.6 release.

Then, if we have `ForUpdateOfSkipLocked rawNames`, we can `mappend` them rather than only taking the last, which would be bad.